DDG4/edm4hep/Geant4Output2EDM4hep.cpp

```diff
@@ -428,9 +444,8 @@ void Geant4Output2EDM4hep::saveCollection(OutputContext<G4Event>& /*ctxt*/, G4VH
   debug("+++ Saving EDM4hep collection %s with %d entries.", colName.c_str(), int(nhits));
   //-------------------------------------------------------------------
   if( typeid( Geant4Tracker::Hit ) == coll->type().type()  ){
-    edm4hep::SimTrackerHitCollection sthc;
     for(unsigned i=0 ; i < nhits ; ++i){
-      auto sth = sthc->create();
+      auto sth = m_trackerHits[colName]->create();
```

Comment:
Can `saveCollection` be called on multiple threads concurrently? In that case the previous version was thread safe, while this will no longer be and we would need to take care of that. The implicit map insertion might be a problem, but also `create` will race when called from multiple threads on the same collection.